src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresAccessPolicyDAO.scala

```diff
@@ -1781,6 +1781,11 @@ class PostgresAccessPolicyDAO(protected val writeDbRef: DbReference, protected v
           from ${ResourceTable as resource}
           join ${ResourceTypeTable as resourceType} on ${resource.resourceTypeId} = ${resourceType.id}
           join ${EffectiveResourcePolicyTable as effPol} on ${resource.id} = ${effPol.resourceId}
+            and ${effPol.resourceId} in (
+              select ${resource.id} from ${ResourceTable as resource}
+              join ${ResourceTypeTable as resourceType} on ${resourceType.id} = ${resource.resourceTypeId}
+              where ${resourceType.name} = ${resourceTypeName}
```

Comment:
if you want to go a little further
```suggestion
              where ${resource.resourceTypeId} = ${resourceTypePKsByName(resourceTypeName)}
```
also could remove the ResourceTypeTable join and replace
```
where ${resourceType.name} = ${resourceTypeName}
```
with
```
where ${resource.resourceTypeId} = ${resourceTypePKsByName(resourceTypeName)}
```